### PR TITLE
Support using local SIFs for container creation

### LINKED ISSUE #344: Support local image registry
Not once we've been asked how to use local SIF images to create pods. Should be considered as enhancement request?

### PR DESCRIPTION
**Pull request by sashayakovtseva**
Closes #344

### CODE REVIEW

File: pkg/image/image.go [hunk 1]
```diff
@@ -98,108 +98,46 @@ func (i *Info) UsedBy() []string {
...
-	if err != nil {
-		return nil, fmt.Errorf("could not open pulled image: %v", err)
-	}
-
-	fi, err := pulled.Stat()
-	if err != nil {
-		return nil, fmt.Errorf("could not fetch file info: %v", err)
+	pullPath := filepath.Join(location, "."+rand.GenerateID(64))
+	glog.V(5).Infof("Pulling %s to temporary file %s", ref, pullPath)
+	cleanup := func() {
```
Comment (cali4888) [inline pkg/image/image.go]: defer -->. if err != nil --> cleanup ?

Comment (sashayakovtseva) [reply pkg/image/image.go]: was trying to get rid of named return values

File: pkg/image/image.go [hunk 2]
```diff
@@ -294,6 +237,92 @@ func (i *Info) Matches(filter *k8s.ImageFilter) bool {
...
+		buildCmd.Stderr = &errMsg
+		buildCmd.Stdout = ioutil.Discard
+		err := buildCmd.Run()
+		if err != nil {
+			return fmt.Errorf("could not build image: %s", &errMsg)
+		}
+	default:
+		return fmt.Errorf("unknown image registry: %s", ref.URI())
+	}
+	return nil
```
Comment (cali4888) [inline pkg/image/image.go]: return fmt.Errorf("unknown image registry: %s", ref.URI())

Comment (cali4888) [reply pkg/image/image.go]: default is not needed

File: pkg/server/image/image.go [hunk 3]
```diff
@@ -80,6 +80,16 @@ func NewSingularityRegistry(storePath string, index *index.ImageIndex) (*Singula
 	return &registry, nil
 }
 
+func (s *SingularityRegistry) Shutdown() error {
```
Comment (cali4888) [inline pkg/server/image/image.go]: comment is missing, I guess

**Pull request merged**